pymc3/distributions/discrete.py

```diff
@@ -603,6 +603,89 @@ def logcdf(value, mu):
         )
 
 
+class GeneralizedPoisson(Discrete):
+    R"""
+    Generalized Poisson log-likelihood.
+
+    Often used to model the number of events occurring in a fixed period
+    of time when the times at which events occur are independent.
+    Offers greater flexibility in disperson than the standard Poisson because
+    the mean and variance do not have to be equal.
+
+    The pmf of this distribution is
+
+    .. math:: f(x \mid \theta, \lambda) = 
+                  \frac{\theta (\theta + \lambda x)^{x-1} e^{-\theta - \lambda x}}{x!}
+
+    ========  ======================================
+    Support   :math:`x \in \mathbb{N}_0`
+    Mean      :math:`\frac{\theta}{1 - \lambda}`
+    Variance  :math:`\frac{\theta}{(1 - \lambda)^3}`
+    ========  ======================================
+
+    Parameters
+    ----------
+    theta: float
+        Mean parameter (theta > 0).
+    lam: float
+        Dispersion parameter (max(-1, -theta/4) <= lam <= 1).
+
+    Notes
+    -----
+    When lam = 0, the Generalized Poisson reduces to the standard Poisson with mu = theta.
+    When lam < 0, the mean is greater than the variance (underdispersion).
+    When lam > 0, the mean is less than the variance (overdispersion).
+    """
+    rv_op = genpoisson
+
+    @classmethod
+    def dist(cls, theta, lam, *args, **kwargs):
+        theta = at.as_tensor_variable(floatX(theta))
+        lam = at.as_tensor_variable(floatX(lam))
+        return super().dist([theta, lam], *args, **kwargs)
+
+    def logp(value, theta, lam):
+        r"""
+        Calculate log-probability of Generalized Poisson distribution at specified value.
+
+        Parameters
+        ----------
+        value: numeric
+            Value(s) for which log-probability is calculated. If the log probabilities for multiple
+            values are desired the values must be provided in a numpy array or Aesara tensor
+
+        Returns
+        -------
+        TensorVariable
+        """
+        log_prob = bound(np.log(theta) + logpow(theta + lam * value, value - 1)
+                         - (theta + lam * value) - factln(value),
+                         theta >= 0,
+                         -1 <= lam, -theta/4 <= lam, lam <= 1,
+                         value >= 0)
+        # Return zero when value > m, where m is the largest pos int for which theta + m * lam > 0 (when lam < 0)
+        return at.switch(theta + value * lam <= 0, 0, log_prob)
+
+    def random(theta, lam, size=None):
```

Comment:
This generator function uses the Inversion Algorithm defined in [this paper](https://www.tandfonline.com/doi/abs/10.1080/01966324.1997.10737439?journalCode=umms20).

I see that the distributions now have a `logcdf` function instead of a `random` function. If it seems like I'm on the right track so far, I'll look into defining a `logcdf` function.

Reply:
The logcdf method is not instead of the random method. Seems comment on the main thread. 

The `logcdf` is also not necessary to implement (but it's a nice plus if you can provide it)